### DISCUSSION TITLE
Conflicting documentation for Jobs/BGTasks iOS

### DISCUSSION THREAD

**Discussion by jmichas** [shinyorg/shiny#655] [Q&A] [answered]
In the Apple documentation for BGTasks they have this statement:
````
Adding a BGTaskSchedulerPermittedIdentifiers key to the Info.plist 
disables application(_:performFetchWithCompletionHandler:) and 
setMinimumBackgroundFetchInterval(_:) in iOS 13 and later.
````
In the shiny documentation there is this:
````
PerformFetch | ShinyPerformFetch
Used by Shiny Core mostly for jobs, but there are lots of places in Shiny where Jobs are used
````
and that we should add the following to info.plist
````
<key>BGTaskSchedulerPermittedIdentifiers</key>
<array>
    <string>com.shiny.job</string>
    <string>com.shiny.jobpower</string>
    <string>com.shiny.jobnet</string>
    <string>com.shiny.jobpowernet</string>
</array>
````

Would this not prevent the `PerformFetch => ShinyPerformFetch` from running on iOS 13+?

**Comment by aritchie** [comment, marked as answer]
Shiny swaps the mechanism internally if BGtasks config is detected.  There was an issue found with the latest version of bgtasks in 2.1.  So you are best to stick with the fetch mechanism or switch to 2.2 preview at this time anyhow.

**Reply by jmichas** [reply]
Ok, so, does 2.2 preview include a fix for rapid fire job on start and resume too? I figured out a work around for the time being but I only run one task because my workaround doesn't scale very easily. 

Jobs in 2.2 use fetch and not BGTasks?

Where can I find 2.2 release notes? Do they exists anywhere?

**Reply by aritchie** [reply]
Yes - that fix is there as well.  

All of Shiny uses BGTasks or Fetch.  2.1 contains a bug so bgtasks isn't working well, so use fetch on 2.1 by disabling "background processing".  2.2 fixes bgtasks but can still work with fetch as well.  

You can find all of the latest release notes on shinylib.net/release-notes
